.circleci/config.yml

```diff
@@ -667,196 +133,19 @@ jobs:
-            - store_artifacts:
-                  path: ~/transformers/reports
-
-# TPU JOBS
-    run_examples_tpu:
-        docker:
-            - image: cimg/python:3.7.12
-        environment:
-            OMP_NUM_THREADS: 1
-            TRANSFORMERS_IS_CI: yes
-        resource_class: xlarge
-        parallelism: 1
-        steps:
-            - checkout
-            - go/install
-            - *checkout_ml_testing
-            - gcp-gke/install
-            - gcp-gke/update-kubeconfig-with-credentials:
-                  cluster: $GKE_CLUSTER
-                  perform-login: true
-            - setup_remote_docker
-            - *build_push_docker
-            - *deploy_cluster
-
-    cleanup-gke-jobs:
-        docker:
-            - image: cimg/python:3.7.12
-        steps:
-            - gcp-gke/install
-            - gcp-gke/update-kubeconfig-with-credentials:
-                  cluster: $GKE_CLUSTER
-                  perform-login: true
-            - *delete_gke_jobs
-
-workflow_filters: &workflow_filters
-    filters:
-        branches:
-            only:
-                - main
 workflows:
     version: 2
-    build_and_test:
+    setup_and_quality:
+        when:
+            not: <<pipeline.parameters.nightly>>
         jobs:
             - check_code_quality
             - check_repository_consistency
             - fetch_tests
-            - run_examples_torch:
-                requires:
-                  - fetch_tests
-            - run_examples_tensorflow:
-                requires:
-                  - fetch_tests
-            - run_examples_flax:
-                requires:
-                  - fetch_tests
-            - run_tests_custom_tokenizers:
-                requires:
-                  - fetch_tests
-            - run_tests_torch_and_tf:
-                requires:
-                  - fetch_tests
-            - run_tests_torch_and_flax:
-                requires:
-                  - fetch_tests
-            - run_tests_torch:
-                requires:
-                  - fetch_tests
-            - run_tests_tf:
-                requires:
-                  - fetch_tests
-            - run_tests_flax:
-                requires:
-                  - fetch_tests
-            - run_tests_pipelines_torch:
-                requires:
-                  - fetch_tests
-            - run_tests_pipelines_tf:
-                requires:
-                  - fetch_tests
-            - run_tests_onnxruntime:
-                requires:
-                  - fetch_tests
-            - run_tests_hub:
-                requires:
-                  - fetch_tests
-            - run_tests_layoutlmv2_and_v3:
-                requires:
-                  - fetch_tests
+
     nightly:
-        triggers:
-            - schedule:
-                cron: "0 0 * * *"
-                filters:
-                    branches:
-                        only:
-                            - main
+        when: <<pipeline.parameters.nightly>>
```

Comment:
Do we need some kind of trigger that will set nightly to `True` in order to run the nightly tests? I see some tutorials mentioning using API to trigger it with specified parameter values. (Or this is already set in our CircleCI project page?)

Reply:
I see you removed the trigger in the last commit - do you have some reference documentation? I can't find anything very clear.

Reply:
The trigger can be setup from the circleCI settings in the Transformers settings (it's actually already launching the job on main every night at midnight UTC but fails right now as the workflow does not have this argument (see [here](https://app.circleci.com/pipelines/github/huggingface/transformers/48853/workflows/72ce7e1a-2c27-44b2-b2bc-cc44495fd3fe)).